app/concerns/locked_document_concern.rb

```diff
@@ -0,0 +1,31 @@
+module LockedDocumentConcern
+  extend ActiveSupport::Concern
+
+  class LockedDocumentError < RuntimeError; end
+
+  def check_if_locked_document_with_content_id(content_id)
+    document = Document.find_by(content_id: content_id)
+    return unless document.present?
+
+    raise_locked_document_error if document.locked?
+  end
+
+  def check_if_locked_document_with_document_id(document_id)
+    document = Document.find(document_id)
+    return unless document.present?
+
+    raise_locked_document_error if document.locked?
+  end
+
+  def check_if_locked_document_with_edition(edition)
+    return unless edition
+
+    raise_locked_document_error if edition.locked?
+  end
```

Comment:
Seeing these with the `with_` suffix makes me think these are looking more like candidates for keyword arguments:

for example:

```
def check_if_locked_document(content_id: nil, edition: nil, document: nil)
  document_locked = document && document.locked?
  edition_locked = edition && edition.document.locked?
  content_id_locked = content_id && Document.exists?(locked: true, content_id: content_id)

  if document_locked || edition_locked || content_id_locked
    raise LockedDocumentError, "Cannot perform this operation on a locked document"
  end 
end
```

I'm also suggesting here switching to use a document model rather than a document_id for consistency with edition - I haven't found a case where this wouldn't meet.